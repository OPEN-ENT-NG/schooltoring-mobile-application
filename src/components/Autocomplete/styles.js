import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  input: {
    backgroundColor: COLORS.WHITE
  },
  list: {
    backgroundColor: COLORS.WHITE,
    zIndex: 10,
    position: "relative"
  },
  item: {
    padding: 20
  }
});
