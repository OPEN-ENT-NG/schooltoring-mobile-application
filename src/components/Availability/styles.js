import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    paddingHorizontal: 30,
    paddingVertical: 15,
    flex: 1,
    justifyContent: "space-evenly"
  },
  item: {
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center",
    height: 60
  },
  itemSwitch: {
    marginRight: 20
  },
  itemText: {
    fontSize: 18
  },
  list: {
    flex: 1,
    justifyContent: "space-around"
  },
  title: {
    fontSize: 18,
    marginBottom: 40,
    flex: 0
  },
  buttonView: {
    flexDirection: "row",
    justifyContent: "flex-end",
    flex: 0,
    padding: 15
  }
});
