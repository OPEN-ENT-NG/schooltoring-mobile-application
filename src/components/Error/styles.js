import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column"
  },
  imageCritical: {
    width: 280,
    height: 301,
    position: "absolute",
    bottom: 0,
    left: -30
  },
  imageUncritical: {
    width: 262,
    height: 301,
    position: "absolute",
    bottom: 0,
    right: -30
  }
});

export default styles;
