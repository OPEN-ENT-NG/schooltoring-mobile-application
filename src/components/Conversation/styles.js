import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

const textColor = {
  color: COLORS.TEXT
};

export default StyleSheet.create({
  conversation: {
    flexDirection: "row",
    paddingHorizontal: 10
  },
  avatar: {
    alignSelf: "flex-start",
    flex: 1,
    alignItems: "center",
    paddingVertical: 20,
    marginHorizontal: 5
  },
  content: {
    flex: 4,
    justifyContent: "space-evenly",
    borderBottomColor: "#000",
    borderBottomWidth: StyleSheet.hairlineWidth
  },
  name: {
    ...textColor,
    fontWeight: "bold",
    fontSize: 20
  },
  message: {
    flexDirection: "row",
    justifyContent: "space-evenly"
  },
  icon: {
    fontSize: 20,
    marginRight: 10
  },
  text: {
    ...textColor,
    flex: 5,
    fontSize: 16,
    flexWrap: "wrap"
  },
  date: {
    flexGrow: 2,
    alignItems: "flex-end"
  },
  dateText: {
    ...textColor
  }
});
