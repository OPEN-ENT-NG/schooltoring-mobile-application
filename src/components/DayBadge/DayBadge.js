import React from "react";
import { View, Text } from "react-native";
import Proptypes from "prop-types";

import styles from "./styles";

const DayBadge = props => (
  <View
    style={[
      styles.item,
      props.available ? styles.available : null,
      props.style
    ]}
  >
    <Text style={styles.itemText}>{props.label}</Text>
  </View>
);

DayBadge.proptypes = {
  label: Proptypes.string.isRequired,
  available: Proptypes.bool.isRequired,
  style: Proptypes.object
};

export default DayBadge;
