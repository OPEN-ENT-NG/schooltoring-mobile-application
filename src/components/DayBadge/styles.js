import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  item: {
    justifyContent: "center",
    alignItems: "center",
    width: 30,
    height: 30,
    backgroundColor: COLORS.GREY
  },
  itemText: {
    flex: 1,
    color: COLORS.WHITE,
    fontSize: 20
  },
  available: {
    backgroundColor: COLORS.NEGATIVE
  }
});
