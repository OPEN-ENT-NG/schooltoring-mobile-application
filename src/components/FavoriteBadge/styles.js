import { StyleSheet, Platform } from "react-native";

import { COLORS } from "../../styles/common";

const shadow =
  Platform.OS === "android"
    ? {
        elevation: 4
      }
    : {
        shadowColor: COLORS.GREY,
        shadowOffset: { width: 2, height: 2 },
        shadowOpacity: 0.6,
        shadowRadius: 1
      };

export default StyleSheet.create({
  badge: {
    alignItems: "stretch",
    width: "50%",
    padding: 10
  },
  top: {
    aspectRatio: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: COLORS.SECONDARY_BACKGROUND,
    ...shadow
  },
  bottom: {
    backgroundColor: COLORS.BACKGROUND,
    ...shadow,
    maxHeight: 120
  },
  name: {
    textAlign: "center",
    color: COLORS.TEXT,
    fontWeight: "bold",
    fontSize: 16
  },
  buttons: {
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: "center",
    padding: 5
  },
  chatButton: {
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: COLORS.PRIMARY
  },
  loader: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: "rgba(255, 255, 255, 0.8)",
    elevation: 1000,
    zIndex: 5
  }
});
