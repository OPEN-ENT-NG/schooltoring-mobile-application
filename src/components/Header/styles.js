import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  header: {
    backgroundColor: COLORS.LIGHT_GREY,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: 10
  },
  titleContainer: {
    flex: 0
  },
  back: { flex: 1 },
  title: {
    fontSize: 25,
    marginLeft: 10
  }
});
