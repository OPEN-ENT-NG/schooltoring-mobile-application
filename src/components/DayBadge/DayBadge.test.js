import React from "react";
import { shallow } from "enzyme";

import DayBadge from "./DayBadge";
import { COLORS } from "../../styles/common";

const label = "L";

const rendered = shallow(<DayBadge label="L" available={true} />);
const renderedNotAvailable = shallow(
  <DayBadge label={label} avaiable={false} />
);
describe("Render", () => {
  test("DayBadge should render without crashing", () => {
    expect(rendered).toMatchSnapshot();
  });

  test(`DayBadge label should render '${label}'`, () => {
    expect(rendered.find("Text").contains(label)).toBeTruthy();
  });

  test(`DayBadge without availability should render background as ${
    COLORS.GREY
  }`, () => {
    expect(renderedNotAvailable.props().style[0].backgroundColor).toEqual(
      COLORS.GREY
    );
  });

  test(`DayBadge with availability should render background as ${
    COLORS.NEGATIVE
  }`, () => {
    expect(rendered.props().style[1].backgroundColor).toEqual(COLORS.NEGATIVE);
  });
});
