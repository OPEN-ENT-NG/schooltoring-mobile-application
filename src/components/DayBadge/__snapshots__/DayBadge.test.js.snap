// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DayBadge should render without crashing 1`] = `
<Component
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#929292",
        "height": 30,
        "justifyContent": "center",
        "width": 30,
      },
      Object {
        "backgroundColor": "#ff6f00",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#ffffff",
        "flex": 1,
        "fontSize": 20,
      }
    }
  >
    L
  </Text>
</Component>
`;
